### PR TITLE
Add kASLR support

### PR DESCRIPTION
**Pull request by ptesarik**
To cope with kASLR, all symbols must be relocated. This is done here using extra arguments to the "add-symbol-file" command.

Note that kASLR also requires a recent version of libaddrxlat for setting up VTOP translation correctly.

### CODE REVIEW

File: crash/session.py [hunk 1]
```diff
@@ -29,7 +30,42 @@ def __init__(self, kernel_exec=None, vmcore=None, kernelpath=None,
...
+                    (addrs, sect) = line.strip().split(" is ")
+                except ValueError:
+                    continue
+                (start, end) = addrs.split(" - ")
+                sect = sect.split(' ')[0]
+                startaddr = long(start, base=0) + kaslr_off
+                if sect == ".text":
+                    textaddr = startaddr
+                    args += " 0x{:x}".format(startaddr)
+                elif startaddr >= textaddr:
```
Comment (Werkov) [inline crash/session.py:61]: What does this condition translate to? In my (non KASLR) ELF it'd only apply to `.data..percpu`.

File: crash/session.py [hunk 2]
```diff
@@ -29,7 +30,42 @@ def __init__(self, kernel_exec=None, vmcore=None, kernelpath=None,
...
+        kaslr_off = long(
+            kdump.attr.get("linux.vmcoreinfo.lines.KERNELOFFSET", "0"),
+            base=16)
+
+        error = gdb.execute("exec-file {}".format(kernel_exec), to_string=True)
+        tinfo = gdb.execute("info target", to_string=True)
+        args=""
+        textaddr = 0
+        in_exec = False
+        for line in tinfo.splitlines():
```
Comment (Werkov) [inline crash/session.py:47]: Probably just  a matter of taste, I'd move the parsing out of `__init__`.

File: crash/session.py (hunk 1 above)
Comment (ptesarik) [reply crash/session.py:61]: Indeed, this serves only the percpu section for now. I'm trying to prepare for a potential rename. The .text section is always the first section in a kernel image, so this looks robust enough to me. Note that the "_stext" symbol is sometimes used to find the beginning of the kernel; it's even exported through VMCOREINFO for that purpose.

File: crash/session.py (hunk 2 above)
Comment (ptesarik) [reply crash/session.py:47]: It looked OK to me for a Hackweek, but if Jeff is also concerned, I can put it in its own method.

**Pull request closed without merging**